### PR TITLE
Standardize naming of AWS Sqs

### PR DESCRIPTION
**Pull request by KennyRich**
Part of #20296

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: CHANGELOG.txt [hunk 1]
```diff
@@ -2591,7 +2591,7 @@ Doc-only changes
 - [AIRFLOW-XXX] update SlackWebhookHook and SlackWebhookOperator docstring (#5074)
 - [AIRFLOW-XXX] Ignore Python files under node_modules in docs (#5063)
 - [AIRFLOW-XXX] Build a universal wheel with LICNESE files (#5052)
-- [AIRFLOW-XXX] Fix docstrings of SQSHook (#5099)
+- [AIRFLOW-XXX] Fix docstrings of SqsHook (#5099)
```
Comment (eladkal) [inline CHANGELOG.txt]: We don't alter past records in change log. this entry was valid to the day it was created.
Please revert this change.

File: UPDATING.md [hunk 2]
```diff
@@ -1994,17 +1994,17 @@ Migrated are:
 | airflow.hooks.S3_hook.S3Hook                                 | airflow.providers.amazon.aws.hooks.s3.S3Hook                    |
 | airflow.contrib.hooks.aws_athena_hook.AWSAthenaHook          | airflow.providers.amazon.aws.hooks.athena.AWSAthenaHook         |
 | airflow.contrib.hooks.aws_lambda_hook.AwsLambdaHook          | airflow.providers.amazon.aws.hooks.lambda_function.AwsLambdaHook         |
-| airflow.contrib.hooks.aws_sqs_hook.SQSHook                   | airflow.providers.amazon.aws.hooks.sqs.SQSHook        |
+| airflow.contrib.hooks.aws_sqs_hook.SqsHook                   | airflow.providers.amazon.aws.hooks.sqs.SqsHook        |
```
Comment (eladkal) [inline UPDATING.md]: There is no `SqsHook` in contrib.
We also can't suggest here `airflow.contrib.hooks.aws_sqs_hook.SQSHook` -> `airflow.providers.amazon.aws.hooks.sqs.SqsHook` because this table is for upgrading from 1.10 to 2.0 and in 2.0 only backport amazon provider is compatible so the changes we do in this PR are not relevant for it.

Long story short - please undo the changes you made for this file :)

File: airflow/contrib/hooks/aws_sqs_hook.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 import warnings
 
-from airflow.providers.amazon.aws.hooks.sqs import SQSHook  # noqa
+from airflow.providers.amazon.aws.hooks.sqs import SqsHook  # noqa
```
Comment (eladkal) [inline airflow/contrib/hooks/aws_sqs_hook.py]: This is not backward compatible. We are changing the class name.
So with the current code when user try to import SQSHook it won't work
Check reference of SESHook https://github.com/apache/airflow/pull/20367/files

It should

File: airflow/contrib/operators/aws_sqs_publish_operator.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
 
 import warnings
 
-from airflow.providers.amazon.aws.operators.sqs import SQSPublishOperator  # noqa
+from airflow.providers.amazon.aws.operators.sqs import SqsPublishOperator  # noqa
```
Comment (eladkal) [inline airflow/contrib/operators/aws_sqs_publish_operator.py]: same

File: airflow/contrib/sensors/aws_sqs_sensor.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 
 import warnings
 
-from airflow.providers.amazon.aws.sensors.sqs import SQSSensor  # noqa
+from airflow.providers.amazon.aws.sensors.sqs import SqsSensor  # noqa
```
Comment (eladkal) [inline airflow/contrib/sensors/aws_sqs_sensor.py]: same

File: airflow/providers/amazon/CHANGELOG.rst [hunk 6]
```diff
@@ -253,7 +253,7 @@ Bug fixes
 * ``Fix 'logging.exception' redundancy (#14823)``
 * ``Fix AthenaSensor calling AthenaHook incorrectly (#15427)``
 * ``Add links to new modules for deprecated modules (#15316)``
-* ``Fixes doc for SQSSensor (#15323)``
+* ``Fixes doc for SqsSensor (#15323)``
```
Comment (eladkal) [inline airflow/providers/amazon/CHANGELOG.rst]: same comment as previous changelog.
Please undo

File: airflow/providers/amazon/aws/hooks/sqs.py [hunk 7]
```diff
@@ -22,7 +22,7 @@
 from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
 
 
-class SQSHook(AwsBaseHook):
+class SqsHook(AwsBaseHook):
```
Comment (eladkal) [inline airflow/providers/amazon/aws/hooks/sqs.py:26]: This is also not backward compatible. We want both `SQSHook` and `SqsHook` to work.

**Review by eladkal — CHANGES_REQUESTED** [review]
Did partial review as some of my comments apply to more than 1 file.

Let me know if something isn't clear enough

File: airflow/contrib/hooks/aws_sqs_hook.py (hunk 3 above)
Comment (KennyRich) [reply airflow/contrib/hooks/aws_sqs_hook.py]: Thanks. I understand that some of the changes aren't backward compatible but, this file has a deprecation warning here already. Do I just revert the changes here and create a deprecation warning for the `SQSHook` and make it backward compatible like this

```
class SQSHook(SqsHook):
    """
        This hook is deprecated.
        Please use :class:`airflow.providers.amazon.aws.hooks.sqs.SqsHook`.
        """

    def __init__(self, *args, **kwargs):
        warnings.warn(
            "This hook is deprecated. " "Please use :class:`airflow.providers.amazon.aws.hooks.sqs.SqsHook`.",
            DeprecationWarning,
            stacklevel=2,
        )
        super().__init__(*args, **kwargs)
```

Comment (eladkal) [reply airflow/contrib/hooks/aws_sqs_hook.py]: yes. The current implementation support deprecation where the class name was not change so it just add an import of where to find it but now -  the class name did change so we must address it. Yes the class you wrote should do the trick.

File: airflow/providers/amazon/aws/hooks/sqs.py (hunk 7 above)
Comment (KennyRich) [reply airflow/providers/amazon/aws/hooks/sqs.py:26]: This has been handled please check and let me know what you think.

File: UPDATING.md (hunk 2 above)
Comment (eladkal) [reply UPDATING.md]: I still see changes on this file
@KennyRich did you undo all changes to this file?

Comment (KennyRich) [reply UPDATING.md]: Yes

File: docs/apache-airflow-providers-amazon/operators/sqs_publish.rst [hunk 8]
```diff
@@ -16,7 +16,7 @@
     under the License.
 
 
-.. _howto/operator:SQSPublishOperator:
+.. _howto/operator:operator:SQSPublishOperator:
```
Comment (eladkal) [inline docs/apache-airflow-providers-amazon/operators/sqs_publish.rst:19]: Is this right ? Shouldn't it be?
```suggestion
.. _howto/operator:SqsPublishOperator:
```

File: docs/apache-airflow-providers-amazon/operators/sqs_publish.rst [hunk 9]
```diff
@@ -34,7 +34,7 @@ Using Operator
 --------------
 
 Use the
-:class:`~airflow.providers.amazon.aws.operators.sqs.SQSPublishOperator`
+:class:`~airflow.providers.amazon.aws.operators.sqs.operator:SQSPublishOperator`
```
Comment (eladkal) [inline docs/apache-airflow-providers-amazon/operators/sqs_publish.rst:37]: same question

File: tests/providers/amazon/aws/operators/test_s3_bucket.py [hunk 10]
```diff
@@ -22,7 +22,7 @@
 from moto import mock_s3
 
 from airflow.providers.amazon.aws.hooks.s3 import S3Hook
-from airflow.providers.amazon.aws.operators.s3 import S3CreateBucketOperator, S3DeleteBucketOperator
+from airflow.providers.amazon.aws.operators.s3_bucket import S3CreateBucketOperator, S3DeleteBucketOperator
```
Comment (eladkal) [inline tests/providers/amazon/aws/operators/test_s3_bucket.py:25]: Why are you modifying s3 files?
I think you have several unrelated changes of s3 in this PR.

**Review by eladkal — CHANGES_REQUESTED** [review]

**Comment by eladkal** [comment]
@KennyRich do you need help to finish this PR?

**Comment by KennyRich** [comment]
> @KennyRich do you need help to finish this PR?

Yes please, I have been unwell for a couple of days now

**Comment by eladkal** [comment]
Closing in favor of https://github.com/apache/airflow/pull/20732
